src/metabase/driver/sql_jdbc/connection.clj

```diff
@@ -215,13 +215,6 @@
   [database]
   (set-pool! (u/the-id database) nil nil))
 
-(defn notify-database-updated
```

Comment:
As mentioned in PR description, this (or rather the function above) adds level of indirection that we are not using currently. I've decided to leave in `invalidate-pool-for-db!` because of more descriptive name.